lib/Sema/CSDiag.cpp

```diff
@@ -6454,6 +6454,50 @@ visitRebindSelfInConstructorExpr(RebindSelfInConstructorExpr *E) {
   return false;
 }
 
+static bool matchArgStructure(Type lhs, Type rhs,
+                              std::function<bool(Type, Type)> comparator) {
```

Comment:
LLVM has a slightly cheaper container for non-escaping callbacks, llvm::function_ref.